### PR TITLE
Replace `if` statement with `if` expression

### PR DESCRIPTION
**Pull request by yezz123**
@CaselIT @zzzeek just some simple enhancements in the code 👌🏻

### CODE REVIEW

**Comment by CaselIT** [comment]
@bourke could you review?

I don't seem to be able to add you to the reviewers in github

**Comment by zzzeek** [comment]
sorry he wasn't in the organization, just sent an invite

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **bourke** to try to get revision dc5ea11602259318b74ad73eb1e28c2f516feb5e of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change dc5ea11602259318b74ad73eb1e28c2f516feb5e: https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

File: mako/codegen.py [hunk 1]
```diff
@@ -524,54 +521,50 @@ def write_variable_declares(self, identifiers, toplevel=False, limit=None):
...
-                    if not comp.is_anonymous:
-                        self.write_def_decl(comp, identifiers)
-                    else:
-                        self.write_inline_def(comp, identifiers, nested=True)
+                if (
+                    comp.is_block
+                    and not comp.is_anonymous
+                    or not comp.is_block
+                    and comp.is_root()
+                ):
```
Comment (sqla-tester) [inline mako/codegen.py:529]: **Michael Bourke** (bourke) wrote:

-1 from me here. I appreciate that it's cleaner in terms of control flow, but it's not as easy to read or understand.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

File: mako/codegen.py [hunk 2]
```diff
@@ -1016,35 +1011,34 @@ class _Identifiers:
     """tracks the status of identifier names as template code is rendered."""
 
     def __init__(self, compiler, node=None, parent=None, nested=False):
-        if parent is not None:
+        if (
+            parent is not None
+            and isinstance(node, parsetree.NamespaceTag)
+            or parent is None
+        ):
             # if we are the branch created in write_namespaces(),
```
Comment (sqla-tester) [inline mako/codegen.py:1019]: **Michael Bourke** (bourke) wrote:

Same as 524-529 above. Looks cleaner, but is harder to understand.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

File: test/__init__.py [hunk 3]
```diff
@@ -131,12 +131,12 @@ def decorate(fn):
...
-            if predicate():
-                msg = "'%s' skipped: %s" % (fn_name, reason)
-                raise unittest.SkipTest(msg)
-            else:
+            if not predicate():
                 return fn(*args, **kw)
 
+            msg = "'%s' skipped: %s" % (fn_name, reason)
+            raise unittest.SkipTest(msg)
+
```
Comment (sqla-tester) [inline test/__init__.py:139]: **Michael Bourke** (bourke) wrote:

Can you help me understand this change? To my eyes, the original expresses the intent more clearly.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

**Review by sqla-tester — COMMENTED** [review]
**Michael Bourke** (bourke) wrote:

Thanks for the contribution; happy to accept it with the following suggestions applied.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

**Comment by bourke** [comment]
Closing this for now. Happy to reopen it if interest revives.

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:



View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/3319 has been **abandoned**.  That means that at least for the moment I need to close this pull request. Sorry it didn't work out :(

**Pull request closed without merging**